src/main/resources/public/jsx/appmanagement/service-settings.jsx.js

```diff
@@ -174,18 +173,18 @@ var ServiceSettings = React.createClass({
         this.setState(state);
     },
     render: function() {
-        var divIconClassName = "form-group";
-        if ($.inArray("icon", this.props.errors) != -1) {
+        let divIconClassName = "form-group";
+        if (this.props.errors.indexOf("icon") !== -1) {
             divIconClassName = divIconClassName + " has-error";
         }
 
-        var visibility = null;
-        if (!this.props.service.service.restricted) {
+        let visibility = null;
+        if (this.props.service.service.access_control !== "ALWAYS_RESTRICTED") {
```

Comment:
This should test `visibility !== "NEVER_VISIBLE"`, and there should (probably) be another checkbox for `access_control`.